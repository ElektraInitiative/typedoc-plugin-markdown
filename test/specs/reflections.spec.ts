import * as Handlebars from 'handlebars';
import { ProjectReflection } from 'typedoc';

describe(`Reflections:`, () => {
  let reflectionTemplate: Handlebars.TemplateDelegate;

  describe(`(header)`, () => {
    let project: ProjectReflection;
    beforeEach(async () => {
      project = await global.bootstrap(['reflections.ts'], {
        hideBreadcrumbs: false,
        hidePageTitle: true,
      });
      global.stubPartials(['comment', 'member.signature', 'members']);
      global.stubHelpers(['toc', 'breadcrumbs', 'hierarchy']);
      reflectionTemplate = global.getTemplate('reflection');
    });
    test(`should compile template with breadcrumbs and without title`, () => {
      expect(
        global.compileTemplate(reflectionTemplate, {
          model: project.children ? project.children[0] : [],
          project: project,
        }),
      ).toMatchSnapshot();
    });
  });

  describe(`(template)`, () => {
    let project: ProjectReflection;
    beforeEach(async () => {
      project = await global.bootstrap(['reflections.ts'], {
        hideBreadcrumbs: true,
        hidePageTitle: false,
      });

      global.stubPartials(['index', 'comment', 'member.signature', 'members']);
      global.stubHelpers(['breadcrumbs', 'hierarchy']);
      reflectionTemplate = global.getTemplate('reflection');
    });

    test(`should compile module with breadcrumbs and project title`, () => {
      expect(
        global.compileTemplate(reflectionTemplate, {
          model: project.children ? project.children[0] : [],
          project: project,
        }),
      ).toMatchSnapshot();
    });

    test(`should compile a callable reflection`, () => {
      expect(
        global.compileTemplate(reflectionTemplate, {
          model: project.getChildByName('CallableReflection'),
          project: project,
        }),
      ).toMatchSnapshot();
    });

    test(`should compile an indexable reflection`, () => {
      expect(
        global.compileTemplate(reflectionTemplate, {
          model: project.getChildByName('IndexableReflection'),
          project: project,
        }),
      ).toMatchSnapshot();
    });

    test(`should compile implemented class`, () => {
      expect(
        global.compileTemplate(reflectionTemplate, {
          model: project.getChildByName('ImplementedClass'),
          project: project,
        }),
      ).toMatchSnapshot();
    });

    test(`should compile Enum`, () => {
      expect(
        global.compileTemplate(reflectionTemplate, {
          model: project.getChildByName('EnumReflection'),
          project: project,
        }),
      ).toMatchSnapshot();
    });
  });
});
