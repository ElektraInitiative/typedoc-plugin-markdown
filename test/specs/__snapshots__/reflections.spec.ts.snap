// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reflections: (header) should compile template with breadcrumbs and without title 1`] = `
"[helper: breadcrumbs]

[helper: toc]

[partial: members]
"
`;

exports[`Reflections: (template) should compile Enum 1`] = `
"# Enumeration: EnumReflection

## Table of contents

### Enumeration Members

- [Down](../enums/EnumReflection.md#down)
- [Left](../enums/EnumReflection.md#left)
- [Right](../enums/EnumReflection.md#right)
- [Up](../enums/EnumReflection.md#up)

[partial: members]
"
`;

exports[`Reflections: (template) should compile a callable reflection 1`] = `
"# Interface: CallableReflection

## Callable

### CallableReflection

[partial: member.signature]

[partial: members]
"
`;

exports[`Reflections: (template) should compile an indexable reflection 1`] = `
"# Interface: IndexableReflection

## Indexable

â–ª [index: \`number\`]: \`string\`

[partial: comment]

[partial: members]
"
`;

exports[`Reflections: (template) should compile implemented class 1`] = `
"# Class: ImplementedClass

[partial: comment]

## Implements

- [\`ReflectionClass\`](../classes/ReflectionClass.md)

## Table of contents

### Constructors

- [constructor](../classes/ImplementedClass.md#constructor)

[partial: members]
"
`;

exports[`Reflections: (template) should compile module with breadcrumbs and project title 1`] = `
"# Enumeration: EnumReflection

## Table of contents

### Enumeration Members

- [Down](../enums/EnumReflection.md#down)
- [Left](../enums/EnumReflection.md#left)
- [Right](../enums/EnumReflection.md#right)
- [Up](../enums/EnumReflection.md#up)

[partial: members]
"
`;
